src/plot_api/plot_api.js

```diff
@@ -153,12 +153,8 @@ Plotly.plot = function(gd, data, layout, config) {
         makePlotFramework(gd);
     }
 
-    if(graphWasEmpty) {
-        // save initial axis range once per graph
-        Plotly.Axes.saveRangeInitial(gd);
-        // save initial show spikes once per graph
-        Plotly.Axes.saveShowSpikeInitial(gd);
-    }
+    // save initial show spikes once per graph
+    if(graphWasEmpty) Plotly.Axes.saveShowSpikeInitial(gd);
```

Comment:
Ah, you already got this merged, great! didn't notice it before making my previous comment obviously :)